pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -326,6 +331,29 @@ func getDeploymentCondition(deployment *appsv1.Deployment, conditionType appsv1.
 	return &condition
 }
 
+// PropagateCrossComponentStatus aggregates the RoutesReady or ConfigurationsReady condition across all available components
+// and propagates the RoutesReady or LatestDeploymentReady status accordingly.
+func (ss *InferenceServiceStatus) PropagateCrossComponentStatus(componentList []ComponentType, conditionType apis.ConditionType) {
+	conditionsMap, ok := conditionsMapIndex[conditionType]
+	if !ok {
+		return
+	}
+	crossComponentCondition := &apis.Condition{
+		Type:   conditionType,
+		Status: v1.ConditionTrue,
+	}
+	for _, component := range componentList {
+		if !ss.IsConditionReady(conditionsMap[component]) {
+			crossComponentCondition.Status = v1.ConditionFalse
+			if ss.IsConditionUnknown(conditionsMap[component]) { // include check for nil condition
+				crossComponentCondition.Status = v1.ConditionUnknown
+			}
+			crossComponentCondition.Reason = string(conditionsMap[component]) + " not ready"
```

Comment:
I think we should aggregate the reasons from individual conditions so we can show the deployment error message